src/main/java/org/openhab/binding/zwave/internal/protocol/ZWaveNode.java

```diff
@@ -1475,27 +1477,41 @@ public void run() {
                 logger.trace("NODE {}: WakeupTimerTask Already asleep", getNodeId());
                 return;
             }
+            count = count + 1;
+            logger.debug("NODE {}: WakeupTimerTask {} Messages waiting, state {} count {}", getNodeId(),
+                    controller.getSendQueueLength(getNodeId()), getNodeInitStage(), count);
 
-            logger.debug("NODE {}: WakeupTimerTask {} Messages waiting, state {}", getNodeId(),
-                    controller.getSendQueueLength(getNodeId()), getNodeInitStage());
-            if (triggered == false) {
-                logger.trace("NODE {}: WakeupTimerTask First iteration", getNodeId());
+            if (isInitializationComplete() == true  && controller.getSendQueueLength(getNodeId()) == 0) {
+                triggered = true;
+            }     
+            if (count == awakemax || count == 40) {
                 triggered = true;
+            }
+            if (triggered == false) {
+                if (count == 4) {
+                    controller.kickQueue();
+                }
+                logger.trace("NODE {}: WakeupTimerTask iteration", getNodeId());
                 return;
             }
 
-            // Tell the device to go back to sleep.
-            logger.debug("NODE {}: No more messages, go back to sleep", getNodeId());
+            // Tell the device to go back to sleep. This message may be queued if the prior message timed out
+            // and the node was therefore assumed to be asleep.  It can disrupt a future awake.
+            if (isInitializationComplete() == true  && controller.getSendQueueLength(getNodeId()) == 0) {
+            logger.debug("NODE {}: Go back to sleep, state {}, count {} messages {}", getNodeId(), getNodeInitStage(), count, controller.getSendQueueLength(getNodeId()));
```

Comment:
Formatting  / indentation seems incorrect around here.

Reply:
Done, (I think)
```
            // Tell the device to go back to sleep. This message may be queued if the prior message timed out
            // and the node was therefore assumed to be asleep.  It can disrupt a future awake.
            if (isInitializationComplete() == true  && controller.getSendQueueLength(getNodeId()) == 0) {
                logger.debug("NODE {}: Go back to sleep, state {}, count {} messages {}", getNodeId(),
                    getNodeInitStage(), count, controller.getSendQueueLength(getNodeId()));
            }
            else {
                logger.info("NODE {}: Wake duration reached, state {} count {}, messages {}", getNodeId(),
                    getNodeInitStage(), count, controller.getSendQueueLength(getNodeId()));
            }
```